src/Compilers/Test/Core/Metadata/MetadataReaderUtils.cs

```diff
@@ -472,11 +473,26 @@ internal static void VerifyPEMetadata(string pePath, string[] types, string[] me
 
                 AssertEx.SetEqual(
                     metadataReader.CustomAttributes.Select(a => metadataReader.GetCustomAttribute(a).Constructor)
-                        .Select(c => metadataReader.GetMemberReference((MemberReferenceHandle)c).Parent)
-                        .Select(p => metadataReader.GetTypeReference((TypeReferenceHandle)p).Name)
+                        .Select(c => getAttributeTypeName(metadataReader, c))
                         .Select(n => metadataReader.GetString(n)),
                     attributes);
             }
+
+            static StringHandle getAttributeTypeName(MetadataReader metadataReader, EntityHandle constructorHandle)
+            {
+                // See MetadataWriter.GetCustomAttributeTypeCodedIndex
+                if (constructorHandle.Kind == HandleKind.MemberReference)
+                {
+                    var typeRef = metadataReader.GetMemberReference((MemberReferenceHandle)constructorHandle).Parent;
+                    return metadataReader.GetTypeReference((TypeReferenceHandle)typeRef).Name;
```

Comment:
>(TypeReferenceHandle)typeRef [](http://example.com/codeflow?start=59&length=28)

Strictly speaking this cast isn't always valid, I think. From spec for MemberRef metadata table: "Class (an index into the MethodDef, ModuleRef,TypeDef, TypeRef, or TypeSpec
tables; more precisely, a MemberRefParent (§II.24.2.6) coded index)." I am not going to block on this because the change is for a test helper, it is an improvement and it is sufficient for the test scenarios we currently have.